umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj

```diff
@@ -16,13 +16,24 @@
   "This is used to check if the constant DOI is contained in the identification description element."
   "DOI")
 
+(defn- is-doi-identified-by-attribute?
+  "Checks to see if the Anchor contains an <gmx:Anchor xlink:title=\"DOI\" attribute. If it does then the
+   identifier is a DOI. Returns true if sucessful or nil if not."
+  [gmd-id]
+  (let [attrs (get (first (select gmd-id "gmd:MD_Identifier/gmd:code/gmx:Anchor")) :attrs)]
+    (some #(when (= :xlink/title (key %))
+             (when (= "DOI" (val %))
```

Comment:
@sxu123 sxu123 37 minutes ago Member
Could you change it to (when (and (= :xlink/title (key %) (= "DOI" (val %))?

@ygliuvt ygliuvt 24 minutes ago Member
No. some takes a function.

Reply:
Right. how about (and  (= :xlink/title (key %)) (= "DOI" (val %)))

Reply:
No. `#` is clojure's way to start an anonymous function. Without `#`, `%` is meaningless.

Reply:
with # of course.

Reply:
I can change the code to look like the following:
    (some #(when (and (= :xlink/title (key %))
                                      (= "DOI" (val %)))
                      true)
               attrs)))

Reply:
I just tested it.

Reply:
I have updated the code to use (some #(and (= :xlink/title (key %) (= "DOI" (val %))) attrs)